bin/aurbuild

```diff
@@ -65,32 +70,50 @@ source /usr/share/makepkg/util.sh
 
 while getopts :a:cd:p:r: OPT; do
     case $OPT in
-        a|+a) queue="$OPTARG"    ;;
-        c|+c) build=build_chroot ;;
-        d|+d) database="$OPTARG" ;;
-        p|+p) pool="$OPTARG"     ;;
-        r|+r) root="$OPTARG"     ;;
-        *) plain "usage: $argv0 [-c] -a <queue> -d <database> -p <pool> -r <root> [--] [makepkg args]"
-           exit 1 ;;
+        a) queue="$OPTARG"    ;;
+        c) build=build_chroot ;;
+        d) database="$OPTARG" ;;
+        p) pool="$OPTARG"     ;;
+        r) root="$OPTARG"     ;;
+        :) case $OPTARG in
+                a) error "$argv0: No package queue specified (-a)"; usage; exit 1 ;;
+                d) error "$argv0: No repository database specified (-d)"; usage; exit 1 ;;
+                p) error "$argv0: No package pool specified (-p)"; usage; exit 1 ;;
+                r) error "$argv0: No repository root specified (-r)"; usage; exit 1 ;;
+            esac ;;
+        ?) error "$argv0: Option not recognized (-$OPTARG)"; usage; exit 1; ;;
     esac
 done
 
+# Validate arguments
+re_blank='^[[:space:]]*$'
+if [[ $queue =~ $re_blank || $database =~ $re_blank || $pool =~ $re_blank || $root =~ $re_blank ]]; then
+    error "$argv0: Missing arguments" && usage
+    exit 1
+elif [[ ! -r $queue ]]; then
```

Comment:
elif lists end on the first succesful test, so in case of multiple wrong arguments, you'd only see the first. A bit more if/fi boilerplate wouldn't hurt here.